### ISSUE TITLE
We can't always combine like skus in the cart

### ISSUE THREAD

**Issue by rmtemplet** [interchange/Interchange6#13]
Customers will sometimes have the need to require that when adding items with the same sku into the cart that it does not combine them into one item. An example of this is from a live IC5 site where you can buy an embosser and personalize it. We also discuss on #interchange the need for this for gift certificates and the ability to personalize those.

http://www.papersource.com/item/Return-Address-Embosser-Serif-Script-Pair-Address/42075100110.html

**Comment by mcmillhj** [comment]
Are you opposed to adding a dependency that supports Union types? Types::Tiny::Union appears to be one such module with no other dependencies. An example of the combine attribute would then look like this: 

```

```

**Comment by SysPete** [comment]
`Type::Tiny` overloads `|` for Union: https://metacpan.org/pod/Type::Tiny#Overloading

So you only need to do:
```
has combine => (
    is      => 'ro'
    isa     => Int | CodeRef,
    default => 1,
);
```
I really like `Type::Tiny` though it can be hard to find stuff like this. I also think perhaps we should use `Bool` rather than `Int` since we don't currently expect int values above 1. If we change our minds later we might add `Enum[]` to specify more complex pre-built conditions.

**Comment by mcmillhj** [comment]
Oh, cool I was not aware that it had that ability. Thanks.

**Comment by SysPete** [comment]
#34 merged locally and pushed. Now need to address update/remove which use sku only and we could have a single sku in multiple cart positions.

**Comment by SysPete** [comment]
Update addressed via commit 66f6f24190da00f489d7984283abca44a6aa9124. Tests needed for this.

**Comment by SysPete** [comment]
Remove addressed via commit b1cef9be935faa30a2229a2f0bf4bf65a0b597e1.

**Comment by SysPete** [comment]
Remove addressed via commit b1cef9be935faa30a2229a2f0bf4bf65a0b597e1 though tests still needed.
